### PR TITLE
Feat : Added Bookmark Feature which lets users bookmark tracks when logged in

### LINKED ISSUE #160: Adding Bookmark button in notes
Users can mark their favorite notes by clicking on a bookmark button. When they want to see these saved notes later, they can find them in the bookmark section of the website's menu, but only if they've logged into their account. If someone tries to mark a note as a favorite without logging in, they'll be asked to log in first through a pop-up message

@hkirat can u assign this issue to me.
I am not familiar with firebase but will sureshot give my best to solve this.

### PR DESCRIPTION
**Pull request by piyushhsainii**
closes #160
- Added Bookmark Feature which lets users bookmark tracks when logged in 

https://github.com/code100x/daily-code/assets/134310734/14d75e70-3b67-4fc6-ba38-7565ab25df66

### CODE REVIEW

File: packages/db/prisma/schema.prisma [hunk 1]
```diff
@@ -49,8 +49,19 @@ model User {
   image         String?
   accounts      Account[]
   sessions      Session[]
+  bookmark      Bookmark[]
 }
 
+model Bookmark {
+  id      String @id @default(uuid())
+  user    String
+  userID  User @relation(fields: [user],references: [id])
```
Comment (Neon-20) [inline packages/db/prisma/schema.prisma:58]: could add onDelete: Cascade on line 58 and 60, to delete the bookmark table if a particular user deletes his/her account.

Comment (piyushhsainii) [reply packages/db/prisma/schema.prisma]: Thanks for the suggestion , totally missed out. I've made the changes now. Let me know if you have any other suggestion :)

File: packages/db/prisma/schema.prisma [hunk 2]
```diff
@@ -66,8 +77,10 @@ model Track {
   image       String
   categories  TrackCategory[] // This replaces 'category Categories[]'
   problems    TrackProblems[]
-  hidden      Boolean        @default(false)
-  createdAt   DateTime       @default(now())
+  hidden      Boolean         @default(false)
+  createdAt   DateTime        @default(now())
+  track       Bookmark[]
```
Comment (hkirat) [inline packages/db/prisma/schema.prisma:82]: whats happening here?

File: apps/web/app/bookmarks/page.tsx [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+import { getAllBookmarkedTrack, getAllTracks } from "../../components/utils";
+import { AppbarClient } from "../../components/AppbarClient";
+import { TrackCard } from "@repo/ui/components";
+import { Session, getServerSession } from "next-auth";
+import { authOptions } from "../../lib/auth";
+
+export const revalidate = 0;
+const page = async () => {
+  const session: Session | null = await getServerSession(authOptions);
+  // @ts-ignore
```
Comment (hkirat) [inline apps/web/app/bookmarks/page.tsx:11]: why the ts ignore?

File: apps/web/components/utils.tsx [hunk 4]
```diff
@@ -94,3 +94,21 @@ export async function getAllCategories() {
     return [];
   }
 }
+
+export async function getAllBookmarkedTrack(userid: string) {
+  try {
+    const bookmarks = await db.bookmark.findMany({
```
Comment (hkirat) [inline apps/web/components/utils.tsx:100]: Paginate it?

File: packages/db/package.json [hunk 5]
```diff
@@ -15,7 +15,7 @@
   "license": "ISC",
   "dependencies": {
     "@prisma/client": "5.12.0",
-    "prisma": "^5.12.0"
+    "prisma": "^5.12.1"
```
Comment (hkirat) [inline packages/db/package.json:18]: Why the update?

**Review by hkirat — CHANGES_REQUESTED** [review]

File: apps/web/.env.example [hunk 6]
```diff
@@ -1,3 +1,11 @@
+GITHUB_ID=""
+GITHUB_SECRET=""
+GOOGLE_CLIENT_ID=""
+GOOGLE_CLIENT_SECRET=""
+NEXTAUTH_URL="http://localhost:3000"
+NEXTAUTH_SECRET="NEXTAUTH_SECRET"
```
Comment (hkirat) [inline apps/web/.env.example:6]: Why dupocate values here?

**Pull request closed without merging**